ExampleMod/Content/CustomModType/CustomModType.cs

```diff
@@ -0,0 +1,353 @@
+
+		/// <summary>
+		/// Called when the pose starts for the given player.
+		/// </summary>
+		public virtual void OnStartPose(Player player) {
+		}
+
+		/// <summary>
+		/// Called each game update while the pose is active.
+		/// </summary>
+		public virtual void Update(Player player) {
+		}
+
+		/// <summary>
+		/// Called when the pose is ending.
+		/// </summary>
+		public virtual void OnEndPose(Player player) {
+		}
+
+		/// <summary>
+		/// Use to control the frame of the texture to display.
+		/// </summary>
+		public virtual Rectangle? GetTextureFrame(Asset<Texture2D> texture) {
+			return null;
+		}
+
+		// TODO: A real implementation might want to add support for conditions or customizable spawn rates
+	}
+
+	public class VictoryPosePlayer : ModPlayer
+	{
+		public ModVictoryPose activeVictoryPose;
+
+		public override void OnHitNPC(NPC target, NPC.HitInfo hit, int damageDone) {
+			// Detect when we defeat an enemy.
+			if (!target.active) {
+				// Start a victory pose
+				StartRandomPose(target.boss);
+			}
+		}
+
+		private void StartRandomPose(bool boss) {
+			// Always pose after defeating a boss, pose 1 out of 100 times for normal enemies.
+			if(!boss && Main.rand.NextBool(99, 100)) {
+				return;
+			}
+
+			if (activeVictoryPose != null) {
+				return; // Don't interrupt an ongoing pose
+			}
+
+			if (boss) {
+				// Choose from all the poses
+				activeVictoryPose = Main.rand.Next(VictoryPoseLoader.VictoryPoses);
+			}
+			else {
+				// Choose from only the NonBoss set
+				int randomNonBossVictoryPoseIndex = Main.rand.Next(VictoryPoseID.Sets.NonBoss.GetTrueIndexes());
+				activeVictoryPose = VictoryPoseLoader.VictoryPoses[randomNonBossVictoryPoseIndex];
+			}
+			// Reset timers.
+			activeVictoryPose.ElapsedPoseTime = 0;
+			activeVictoryPose.OnStartPose(Player);
+
+			ChatHelper.DisplayMessage(activeVictoryPose.VictoryCheer.ToNetworkText(), Color.White, (byte)Main.myPlayer);
+
+			Asset<Texture2D> texture = ModContent.Request<Texture2D>(activeVictoryPose.Texture);
+			Rectangle? frame = activeVictoryPose.GetTextureFrame(texture);
+			Main.ParticleSystem_World_BehindPlayers.Add(new PoseIconParticle(texture, frame, new Vector2(Main.rand.NextFloat(-2, 2), -5f), Player.Center - new Vector2(0, 80), Main.rand.NextFloat(-0.03f, 0.03f)) {
+				AccelerationPerFrame = new Vector2(0f, 0.16350001f),
+				ScaleOffsetPerFrame = 1f / 60f,
+			});
+
+			// TODO: A real implementation would want to sync the pose to other clients with a ModPacket
+		}
+
+		public override void PostUpdate() {
+			// Here we manage the lifetime of a victory pose
+			if (activeVictoryPose == null) {
+				return;
+			}
+
+			activeVictoryPose.Update(Player);
+			activeVictoryPose.ElapsedPoseTime++;
+			if (activeVictoryPose.ElapsedPoseTime >= activeVictoryPose.PoseTime) {
+				activeVictoryPose.OnEndPose(Player);
+				activeVictoryPose = null;
+			}
+		}
+	}
+
+	public class VictoryPoseLoader : ILoadable
```

Comment:
I think it best to split all classes here to their own `.cs` files, as long as they remain contained in the one folder